### DISCUSSION TITLE
KeePass XC Mac - How to find and remove duplicate entries?

### DISCUSSION THREAD

**Discussion by DaveLeonard** [keepassxreboot/keepassxc#8420] [Q&A] [answered]
Hello Folks,

I'm looking for a way to cleanup my KeePass Database from multiple identical duplicated entries.
Is there a way to do that on KeePass Mac?

Thank you very much?

Best, David

**Comment by droidmonkey** [comment, marked as answer]
Search for `*` then sort by whatever column you want to find duplicates.

**Reply by DaveLeonard** [reply]
Hey @droidmonkey, thanks for the idea, but unfortunately I have thousands of duplicates to select.
I'm looking for an automated way to select one of every duplicate entry to remove it.

**Reply by phoerious** [reply]
Sounds like you should be using keepassxc-cli.

**Reply by DaveLeonard** [reply]
Hey @phoerious, thanks. I did look into it, but I'm don't see how it could be done with the cli

**Reply by phoerious** [reply]
Write a script that uses the cli to list entries, find duplicates and then merge/delete them.

**Reply by DaveLeonard** [reply]
Ok thanks

**Reply by djcarryout** [reply]
Does anyone know of an example script i could use as a starting point of doing this? I have the same need as the original poster. Otherwise i will try to write one later and post it here. Thanks!

**Reply by DaveLeonard** [reply]
hey @djcarryout, don't have a script but interested in yours.
